assets/js/modules/analytics/datastore/report.js

```diff
@@ -56,6 +58,7 @@ const fetchGetReportStore = createFetchStore( {
 
 		const { metrics, dimensions, orderby } = options;
 
+		invariant( !! metrics, 'Requests must specify at least one metric for an Analytics report.' );
```

Comment:
The API requires at least one metric to be requested so we should enforce that here, rather than in `isValidMetrics` which more about the shape than request specific validation IMO.

Reply:
`isValidMetrics` mostly checks the types/shape, it doesn't ensure that a non-empty value isn't provided. It is also a more useful error I think if no metrics were provided at all. Without this it could be called with `metrics: ''` and pass the validation.

Taking a closer look at `isValidStringsOrObjects`, it accepts any string (including empty), as well as an empty array, but as a generic utility, it isn't clear if these should be considered invalid or not. Also the `verifyFunction` is only called for objects. It might make more sense to split this into `areValidStrings` and `areValidObjects` functions. Then these aren't even really separate functions anymore, and you can have a simple `isValidCollection` function that looks something like this: `castArray( collection ).every( validateFn )`, but you still end up needing to validate the list length (whether or not an empty list should be valid). For that reason, maybe we should just normalize first here, ensure the list isn't empty and then validate each item in the list?